WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/dashboard/activity/ActivityCardBuilder.kt

```diff
@@ -0,0 +1,73 @@
+import org.wordpress.android.ui.activitylog.list.ActivityLogListItem
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.DashboardCards.DashboardCard.ActivityCard
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.DashboardCards.DashboardCard.ActivityCard.ActivityCardWithItems
+import org.wordpress.android.ui.mysite.MySiteCardAndItem.Card.DashboardCards.DashboardCard.ActivityCard.ActivityCardWithItems.ActivityItem
+import org.wordpress.android.ui.mysite.MySiteCardAndItemBuilderParams.ActivityCardBuilderParams
+import org.wordpress.android.ui.utils.ListItemInteraction
+import org.wordpress.android.ui.utils.UiString
+import org.wordpress.android.ui.utils.UiString.UiStringRes
+import org.wordpress.android.util.DateTimeUtilsWrapper
+import org.wordpress.android.util.SiteUtilsWrapper
+import org.wordpress.android.util.config.DashboardCardActivityLogConfig
+import java.util.Date
+import javax.inject.Inject
+
+private const val MAX_ITEMS_IN_CARD: Int = 3
+
+class ActivityCardBuilder @Inject constructor(
+    private val dashboardCardActivityLogConfig: DashboardCardActivityLogConfig,
+    private val dateTimeUtilsWrapper: DateTimeUtilsWrapper,
+    private val siteUtilsWrapper: SiteUtilsWrapper,
+) {
+    fun build(params: ActivityCardBuilderParams): ActivityCard? {
+        return if (shouldBuildActivityCard(params)) {
+            buildActivityCard(params)
+        } else {
+            null
+        }
+    }
+
+    private fun buildActivityCard(params: ActivityCardBuilderParams): ActivityCardWithItems {
+        val activities = params.activityCardModel?.activities
+        val content = activities?.take(MAX_ITEMS_IN_CARD)?.mapToActivityItems(params.onActivityItemClick) ?: emptyList()
+        return ActivityCardWithItems(
+            title = UiStringRes(R.string.dashboard_activity_card_title),
+            activityItems = content,
+            footerLink = ActivityCard.FooterLink(
+                label = UiStringRes(R.string.dashboard_activity_card_footer_link),
+                onClick = params.onFooterLinkClick
+            )
+        )
+    }
+
+    private fun List<ActivityLogModel>.mapToActivityItems(onClick: (activityId: String) -> Unit) =
+        map {
+            ActivityItem(
+                title =  UiString.UiStringText(it.summary),
+                content = it.content?.text?:"",
+                displayDate = buildDateLine(it.published),
+                icon = ActivityLogListItem.Icon.fromValue(it.gridicon).drawable,
+                iconBackgroundColor = ActivityLogListItem.Status.fromValue(it.status).color,
+                onClick = ListItemInteraction.create(it.activityID, onClick),
+            )
+        }
+
+    private fun buildDateLine(published: Date) = dateTimeUtilsWrapper.javaDateToTimeSpan(published)
+
+    private fun shouldBuildActivityCard(params: ActivityCardBuilderParams) : Boolean {
+        if (!dashboardCardActivityLogConfig.isEnabled() ||
+            params.activityCardModel == null ||
```

Comment:
I use AS "Reformat Code", could be I have different styles set. 🤷  I can go either way with this.

Reply:
I asked just for my knowledge, I'm fine with because my only argument to favor the other way when one is asking which way to go is:
- we're reading left to right 😅